### PR TITLE
Make SeltzerWebSocket delegate to WebSocketChannel

### PR DESCRIPTION
**Pull request by kendalharland**
- Change futures on sendX and close() to void, since Future adds nothing.
- Clean up tests for close()
- major version bump

### CODE REVIEW

File: lib/platform/browser.dart [hunk 1]
```diff
@@ -13,7 +15,8 @@ export 'package:seltzer/seltzer.dart';
 /// This is appropriate for clients running in Dartium, DDC, and dart2js.
 void useSeltzerInTheBrowser() {
   setHttpPlatform(const BrowserSeltzerHttp());
-  setSocketPlatform(BrowserSeltzerWebSocket.connect);
+  setSocketPlatform((String url) async =>
+      new SeltzerWebSocketImpl(new HtmlWebSocketChannel.connect(url)));
```
Comment (matanlurey) [inline lib/platform/browser.dart]: async?

File: lib/platform/vm.dart [hunk 2]
```diff
@@ -13,7 +15,8 @@ export 'package:seltzer/seltzer.dart';
 /// This is appropriate for clients running in the VM on the command line.
 void useSeltzerInVm() {
   setHttpPlatform(const VmSeltzerHttp());
-  setSocketPlatform(VmSeltzerWebSocket.connect);
+  setSocketPlatform((String url) async =>
```
Comment (matanlurey) [inline lib/platform/vm.dart]: Ditto, async?

File: lib/src/socket_impl.dart [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import 'dart:async';
+import 'dart:typed_data';
+
+import 'package:async/async.dart';
+import 'package:seltzer/src/interface.dart';
+import 'package:web_socket_channel/web_socket_channel.dart';
+
+/// A cross-platform [SeltzerWebSocket] implementation.
+class SeltzerWebSocketImpl implements SeltzerWebSocket {
```
Comment (matanlurey) [inline lib/src/socket_impl.dart]: How about `ChannelWebSocket`?

File: lib/src/socket_impl.dart [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  final Completer<SeltzerSocketClosedEvent> _onCloseCompleter =
+      new Completer<SeltzerSocketClosedEvent>();
+  final StreamSplitter<SeltzerMessage> _messageStreamSplitter;
+  final WebSocketChannel _delegate;
+
+  bool _isOpen = true;
+
+  /// Creates a [SeltzerWebSocketImpl] that communicates through [channel].
+  SeltzerWebSocketImpl(WebSocketChannel channel)
```
Comment (matanlurey) [inline lib/src/socket_impl.dart]: Can you make sure to use `@override` here when needed?

File: pubspec.yaml [hunk 5]
```diff
@@ -1,5 +1,5 @@
 name: seltzer
-version: 0.2.4-alpha
+version: 1.2.4-alpha
```
Comment (matanlurey) [inline pubspec.yaml]: In pre-1.0.0, it's not necessary to make a jump like this.

Just go ahead and bump to `0.3.0`.

**Review by matanlurey — CHANGES_REQUESTED** [review]

File: lib/platform/browser.dart (hunk 1 above)
Comment (kendalharland) [reply lib/platform/browser.dart]: done

File: lib/platform/vm.dart (hunk 2 above)
Comment (kendalharland) [reply lib/platform/vm.dart]: done

File: lib/src/socket_impl.dart (hunk 3 above)
Comment (kendalharland) [reply lib/src/socket_impl.dart]: done

File: lib/src/socket_impl.dart (hunk 4 above)
Comment (kendalharland) [reply lib/src/socket_impl.dart]: done

File: pubspec.yaml (hunk 5 above)
Comment (kendalharland) [reply pubspec.yaml]: done

**Review by matanlurey — APPROVED** [review]

**Pull request merged**